### PR TITLE
DSD-440: form spacing variants

### PR DESCRIPTION
**Pull request by bigfishdesign13**
Fixes JIRA ticket [DSD-440](https://jira.nypl.org/browse/DSD-440)

## This PR does the following:
- adds `FormSpacing` enum
- adds `spacing` prop to `Form` component

## How has this been tested?

- DS local build

### Checklist:

- [x] I have updated the Storybook documentation accordingly.
- [x] All new and existing tests passed.

### Front End Review:

- [x] View [the example in Storybook](https://pr651-jrksmtutj1qb2jjv9ht2p2owihilhlbl.tugboat.qa/)

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr651-jrksmtutj1qb2jjv9ht2p2owihilhlbl.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/611fe0ba378c5f45ca85d087

File: src/components/Form/Form.stories.mdx [hunk 1]
```diff
@@ -142,13 +149,83 @@ The `Form` component renders a standard `<form>` element and should be used to h
...
 
-```
-<Form
-  action="/end/point"
-  method="get"
->
+export const formRow = (size) => {
+  const labelText = `Size: ${size}`;
+  return (
+    <>
```
Comment (EdwinGuzman) [inline src/components/Form/Form.stories.mdx]: This should be an `li`.

**Review by EdwinGuzman — CHANGES_REQUESTED** [review]
Just a minor update but everything else looks okay.

File: src/components/Form/Form.stories.mdx (hunk 1 above)
Comment (bigfishdesign13) [reply src/components/Form/Form.stories.mdx]: Yes.  Got catch.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr651-jrksmtutj1qb2jjv9ht2p2owihilhlbl.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/611fe0ba378c5f45ca85d087

**Review by EdwinGuzman — APPROVED** [review]

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr651-jrksmtutj1qb2jjv9ht2p2owihilhlbl.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/611fe0ba378c5f45ca85d087

**Pull request merged**